python-venv/cmstools/test/cfs_sessions_rc_test/__main__.py

```diff
@@ -0,0 +1,151 @@
+#
+# MIT License
+#
+# (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+import argparse
+import sys
+from typing import NoReturn
+
+from cmstools.test.cfs_sessions_rc_test.helpers.cleanup import cleanup_and_restore
+from cmstools.test.cfs_sessions_rc_test.helpers.setup import get_cfs_config_name
+from cmstools.test.cfs_sessions_rc_test.helpers.setup import cfs_sessions_rc_test_setup
+from cmstools.test.cfs_sessions_rc_test.subtests import all_subtests
+from cmstools.lib.log import LOG_FILE_PATH
+
+from .log import logger
+from .defs import ScriptArgs, CFSRCException
+from .argument_parser import (add_mutually_exclusive_arguments, add_setup_arguments,
+                              add_subtests_arguments, add_cfs_session_arguments)
+
+
+def get_test_names(script_args: ScriptArgs) -> list[str]:
+    """Get the list of subtests to run or skip based on command line arguments."""
+    all_tests = list(all_subtests().keys())
+
+    if script_args.run_subtests:
+        return script_args.run_subtests
+
+    if script_args.skip_subtests:
+        return [name for name in all_tests if name not in script_args.skip_subtests]
+
+    return all_tests
```

Comment:
Not a big deal, but I'd recommend a few small changes here.

1. No need to call `all_subtests()` until after you know you will need it.

2. When you call it, you can just store it as-is, as a dict. In the list comprehension on line 49, when it iterates over the dict, it will be iterating over its keys.

```
>>> mydict = { "key1": "value1", "key2": "value2" }
>>> mylist_comp = [ x for x in mydict ]
>>> print(mylist_comp)
['key1', 'key2']
```

3. Only when you return it at the end of the function do you care about turning it into a list, since that is what the function is supposed to return. But for that, the call to `.keys()` is unnecessary -- like we saw before, that is the default behavior when iterating over a dict (which is what it does in the case of making it into a list).

```
>>> mylist = list(mydict)
>>> print(mylist)
['key1', 'key2']
```

```suggestion
    if script_args.run_subtests:
        return script_args.run_subtests

    all_tests = all_subtests()

    if script_args.skip_subtests:
        return [name for name in all_tests if name not in script_args.skip_subtests]

    return list(all_tests)
```